sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/SchemaTransform.java

```diff
@@ -33,5 +38,60 @@
  * compatibility guarantees and it should not be implemented outside of the Beam repository.
  */
 @Internal
-public abstract class SchemaTransform
-    extends PTransform<PCollectionRowTuple, PCollectionRowTuple> {}
+public abstract class SchemaTransform extends PTransform<PCollectionRowTuple, PCollectionRowTuple> {
+  private @Nullable Row configurationRow;
+  private @Nullable String identifier;
+  private boolean registered = false;
```

Comment:
I use "registered" throughout, but I'm not sure if it's the best terminology to use in this context. Open to other suggestions!